sklego/meta/grouped_predictor.py

```diff
@@ -47,13 +50,18 @@ def __init__(
         groups,
         shrinkage=None,
         use_global_model=True,
+        check_X=True,
         **shrinkage_kwargs,
     ):
         self.estimator = estimator
         self.groups = groups
         self.shrinkage = shrinkage
         self.use_global_model = use_global_model
         self.shrinkage_kwargs = shrinkage_kwargs
+        self.check_X = check_X
```

Comment:
I see a slight issue with how this is implemented. When the estimator is cloned, the `set_params` won't set the `check_X` value of the `_check_kwargs` dict. I'd suggest not going through the `_check_kwargs` dictionary and instead add `self.check_X` explicitly to the calls to `_split_groups_and_values`

Reply:
I agree with your suggestion and will push new commits soon. I apologize for not understanding and taking into account the mechanism of `set_params` here.

Reply:
No worries! It's probably the most confusing bit about writing sklearn extensions